### ISSUE TITLE
Question about labels processing

### ISSUE THREAD

**Issue by Amber0913** [eric-ai-lab/MiniGPT-5#58]
Hello, thank you very much for providing the detailed training code. I have a question about the handling of labels in the dataloader. 

During training, 8 image identifiers ([IMG0] to [IMG7]) are introduced, and the labels initially obtain the ids corresponding to these 8 identifiers. However, the subsequent code processes the labels, retaining only the ids corresponding to the first and last identifiers ([IMG0] and [IMG7]), while setting the labels in between to -100. Meanwhile, the hidden_states corresponding to these 8 identifiers, with a shape of [1, 8, dim], are further processed and eventually input into the SD model.

My confusion is: 
1. Why are the labels in the middle processed this way? Is it to retain the identifiers at both ends to locate the positions of the image tokens? 
2. But if so, why not just keep the first identifier? The last one also seems convertible to -100. 
3. Also, why aren't the hidden_​​states (of shape [1, 6, dim]) corresponding to the middle 6 identifiers further processed and fed into the SD model later? Wouldn't the hidden_​​states corresponding to the first and last identifiers affect the generation of the SD?

My question may be a bit stupid, I hope you can help me answer it.

```
        if len(index):
            index = index[0,1]
            print("index: {}".format(index))
            label[:, index+1:index+IMG_TOKEN_NUM-1] = -100
```

**Comment by KzZheng** [comment]
To answer your questions:
1. Yes, they are retrained as beginning and end for the location of the positions of image tokens.
2. We also hope the model can learn the last image token positions so that LLM knows where to continue output text tokens.
3. I think you can also try to use only the middle hidden states. We didn't train on that way but it should also work well.

Thanks for your attention on our works!

**Comment by Amber0913** [comment]
It's clear to me now, thank you very much for your reply!
